test/Facade.test.ts

```diff
@@ -219,819 +203,917 @@ describe('FacadeRead + FacadeAct contracts', () => {
+        const [newToks, newQuantities, newAvailable] = await facadeMint.callStatic.redeem(
+          rToken.address,
+          issueAmount
+        )
+        expect(newToks[0]).to.equal(token.address)
+        expect(newQuantities[0]).to.equal(issueAmount.div(4))
+        expect(newAvailable[0]).to.equal(issueAmount.div(4).div(2))
 
-      // Redeem all RTokens
-      await rToken.connect(addr1).redeem(issueAmount)
+        // redeemCustom
+        const [newToksCustom, newQuantitiesCustom] = await facadeMint.callStatic.redeemCustom(
+          rToken.address,
+          issueAmount,
+          [await basketHandler.nonce()],
+          [fp('1')]
+        )
+        expect(newToksCustom.length).to.equal(4)
+        expect(newToksCustom[0]).to.equal(token.address)
+        expect(newToksCustom[1]).to.equal(usdc.address)
+        expect(newToksCustom[2]).to.equal(aToken.address)
+        expect(newToksCustom[3]).to.equal(cTokenVault.address)
+        expect(newQuantitiesCustom[0]).to.equal(issueAmount.div(4).div(2))
+        expect(newQuantitiesCustom[1]).to.equal(issueAmount.div(4).div(bn('1e12')))
+        expect(newQuantitiesCustom[2]).to.equal(issueAmount.div(4))
+        expect(newQuantitiesCustom[3]).to.equal(issueAmount.div(4).mul(50).div(bn('1e10')))
+
+        // refreshBasket()
+        await basketHandler.connect(owner).refreshBasket()
+        await expect(facadeMint.callStatic.redeem(rToken.address, issueAmount)).not.to.be.reverted
+        const [prevBasketTokens, prevBasketQuantities] = await facadeMint.callStatic.redeemCustom(
+          rToken.address,
+          issueAmount,
+          [(await basketHandler.nonce()) - 1],
+          [fp('1')]
+        )
+        expect(prevBasketTokens.length).to.equal(4)
+        expect(prevBasketTokens[0]).to.equal(token.address)
+        expect(prevBasketTokens[1]).to.equal(usdc.address)
+        expect(prevBasketTokens[2]).to.equal(aToken.address)
+        expect(prevBasketTokens[3]).to.equal(cTokenVault.address)
+        expect(prevBasketQuantities[0]).to.equal(issueAmount.div(4).div(2))
+        expect(prevBasketQuantities[1]).to.equal(issueAmount.div(4).div(bn('1e12')))
+        expect(prevBasketQuantities[2]).to.equal(issueAmount.div(4))
+        expect(prevBasketQuantities[3]).to.equal(issueAmount.div(4).mul(50).div(bn('1e10')))
+      })
 
-      // Check values = 0 (no supply)
-      ;[backing, overCollateralization] = await facade.callStatic.backingOverview(rToken.address)
+      it('Should revert when returning redeemable quantities if frozen', async () => {
+        await main.connect(owner).freezeShort()
+        await expect(facadeMint.callStatic.redeem(rToken.address, issueAmount)).to.be.revertedWith(
+          'frozen'
+        )
+      })
 
-      // Check values - No supply, returns 0
-      expect(backing).to.equal(0)
-      expect(overCollateralization).to.equal(0)
+      itP1('Should calculate customRedeem portions correctly', async () => {
```

Comment:
seems like we should also test at least 1 or 2 scenarios that are a little more complex than this?